jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+        """
+        
+        amp_meds = {}
+
+        for key in amps:
+            x_start, x_end, x_step = amps[key][0]
+            y_start, y_end, y_step = amps[key][1]
+            
+            # Find median value of both even and odd columns for this amp
+            amp_med_even = np.nanmedian(image[y_start: y_end, x_start: x_end][:, 1::2])
+            amp_meds['amp{}_even_med'.format(key)] = amp_med_even
+            amp_med_odd = np.nanmedian(image[y_start: y_end, x_start: x_end][:, ::2])
+            amp_meds['amp{}_odd_med'.format(key)] = amp_med_odd
+
+        return amp_meds
+
+    def identify_tables(self):
+        """Determine which database tables to use for a run of the bias 
+        monitor.
+        """
+
+        mixed_case_name = JWST_INSTRUMENT_NAMES_MIXEDCASE[self.instrument]
+        self.query_table = eval('{}BiasQueryHistory'.format(mixed_case_name))
+        self.stats_table = eval('{}BiasStats'.format(mixed_case_name))
+
+    def image_to_png(self, image, outname):
+        """Ouputs an image array into a png file.
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            2D image array
+
+        outname : str
+            The name given to the output png file
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the output png file
+        """
+
+        output_filename = os.path.join(self.data_dir, '{}.png'.format(outname))
+
+        if not os.path.isfile(output_filename):
+            # Get image scale limits
+            z = ZScaleInterval()
+            vmin, vmax = z.get_limits(image)
+
+            # Plot the image
+            plt.figure(figsize=(12,12))
+            ax = plt.gca()
+            im = ax.imshow(image, cmap='gray', origin='lower', vmin=vmin, vmax=vmax)
+            ax.set_title('{}'.format(outname))
+
+            # Make the colorbar
+            divider = make_axes_locatable(ax)
+            cax = divider.append_axes("right", size="5%", pad=0.4)
+            cbar = plt.colorbar(im, cax=cax)
+            cbar.set_label('Signal [DN]')
+
+            plt.savefig(output_filename, bbox_inches='tight', dpi=200)
+            logging.info('\t{} created'.format(output_filename))
+        else:
+            logging.info('\t{} already exists'.format(output_filename))
+            pass
+
+        return output_filename
+
+    def most_recent_search(self):
+        """Query the query history database and return the information
+        on the most recent query for the given ``aperture_name`` where
+        the bias monitor was executed.
+
+        Returns
+        -------
+        query_result : float
+            Date (in MJD) of the ending range of the previous MAST query
+            where the bias monitor was run.
+        """
+
+        sub_query = session.query(self.query_table.aperture,
+                                  func.max(self.query_table.end_time_mjd).label('maxdate')
```

Comment:
I didn't know you could assign a `.label()`, neat!

Reply:
thanks - I stole it from Bryan's code haha